Sources/Puree/Filter.swift

```diff
@@ -2,13 +2,22 @@ import Foundation
 
 public typealias FilterOptions = [String: Any]
 
-public struct FilterSetting {
+public protocol FilterSettingProtocol {
+    func makeFilter() throws -> Filter
```

Comment:
Wouldn't the following make things simpler and cleaner?
```swift
public protocol FilterSettingProtocol {
    var filterMaker: () -> Filter { get }
}
```

Reply:
After thinking more about it, it seems to me that `FilterSetting` is not even useful. Directly passing a `Filter` to `Configuration` would achieve the same result...

Reply:
I agree. I'll modify this interface next major version.